### PR TITLE
[NativeAOT-LLVM] Migrate the last LLVM helper to WASM and delete `LLVMObjectWriter`

### PR DESCRIPTION
**Pull request by SingleAccretion**
Continuation of https://github.com/dotnet/runtimelab/pull/2645.

No-diff change.

### CODE REVIEW

**Comment by SingleAccretion** [comment]
@dotnet/nativeaot-llvm

File: src/coreclr/tools/aot/ILCompiler.LLVM/Compiler/DependencyAnalysis/ExternMethodAccessorNode.cs [hunk 1]
```diff
@@ -67,48 +71,121 @@ public IEnumerable<MethodDesc> EnumerateMethods()
...
 
-                default:
-                    foreach (MethodDesc method in (HashSet<MethodDesc>)_methods)
-                    {
-                        return method;
-                    }
+            // TODO-LLVM: the initial design of ExternMethodAccessorNode assumed that, eventually, ILC would not
+            // need to concern itself with ABI (i. e. that only the Jit would need to know the ABI details). That
+            // is why Signature is provided by Jit. However, at this point, it has become clear that we will not
+            // be able to avoid WASM signature building in ILC, and so the Jit-based mechanism is basically (ILC
```
Comment (yowl) [inline src/coreclr/tools/aot/ILCompiler.LLVM/Compiler/DependencyAnalysis/ExternMethodAccessorNode.cs]: missing a word, "redundant"?

**Review by yowl — COMMENTED** [review]
Good stuff, just a comment nit

File: src/coreclr/tools/aot/ILCompiler.LLVM/Compiler/DependencyAnalysis/ExternMethodAccessorNode.cs (hunk 1 above)
Comment (SingleAccretion) [reply src/coreclr/tools/aot/ILCompiler.LLVM/Compiler/DependencyAnalysis/ExternMethodAccessorNode.cs]: Yep. Fixed.

**Comment by yowl** [comment]
This is failing with the libz problem, it can be taken from https://github.com/dotnet/runtime/issues/106566 or merged after #2605.

**Review by yowl — APPROVED** [review]

**Comment by jkotas** [comment]
Thank you!

**Pull request merged**